internal/log_analysis/log_processor/pantherlog/scanners.go

```diff
@@ -190,3 +190,37 @@ func ScanEmail(w ValueWriter, input string) {
 		w.WriteValues(FieldEmail, input)
 	}
 }
+
+func ScanMD5Hash(w ValueWriter, input string) {
+	input = strings.TrimSpace(input)
+	if len(input) == 32 && isHex(input) {
+		w.WriteValues(FieldMD5Hash, input)
+	}
+}
+
+func ScanSHA1Hash(w ValueWriter, input string) {
+	input = strings.TrimSpace(input)
+	if len(input) == 40 && isHex(input) {
+		w.WriteValues(FieldSHA1Hash, input)
+	}
+}
+
+func ScanSHA256Hash(w ValueWriter, input string) {
+	input = strings.TrimSpace(input)
+	if len(input) == 64 && isHex(input) {
+		w.WriteValues(FieldSHA256Hash, input)
+	}
+}
+
+func isHex(s string) bool {
+	if len(s) == 0 {
+		return false
+	}
+	for _, c := range s {
```

Comment:
A `range` over `s` [will go through UTF8 decoding and iterate over `rune` values](https://golangbot.com/strings/). It's best to use a normal `i` loop to iterate over bytes in this case since hex strings are ASCII. 
```suggestion
	for i := 0; 0 <= i && i < len(s); i++ {
	  if !isHexDigit(s[i]) {
	    return false
	  }
	}
	return true
  }
// this function gets inlined by the compiler
// we use  a function to make the check in the loop above more readable
func isHexDigit(c byte) bool {
  return ('A' <= c && c <= 'F') || ('a' <= c && c <= 'f') || ('0' <= c && c <= '9')
}
```

Reply:
Good point.

Reply:
Done